### DISCUSSION TITLE
Server error: ERROR While handling GETPORTNUMBERS request [WinError 10048]

### DISCUSSION THREAD

**Discussion by bradut** [GerryDazoo/Slinger#589] [General]
Hi,
I've tried to connect to a slingbox_server which was intensely used until two days ago.
I don't know if the person who used it noticed any error, but today I've seen that the player indicated that something was wrong with the server.

Indeed, the server's console was filled with this message:
`ERROR While handling GETPORTNUMBERS request [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted`

In the Slinger Player's remote control window I clicked on the _Restart_ button, and the only effect was that its web page could no longer be reloaded. No effect on server's console.

I then reset the server, while the player was still open on my side, and the server complained about incorrect request for remote and indicated some python code lines, something like this:
```
05/02/2023, 05:36:06.672   RemoteControl connection from ('...)
Remote Control Connected
GET Remote
Error: Remote URL needs to be one of:
/Remote/sb1
/Remote/sb2
.....
+ some py errors I forgot to capture :)
```

Finally, after restarting the player and the server, everything was fine (and Restart stopped the server successfully).

To address these issues, I would like to suggest implementing some features in the server:
- retry logic 
  - when all available ports on the machine are exhausted. 
  - when an incorrect remote URL is passed
  - ...
- restart the server when restart event is sent instead of turning it off.
- option to log events into a file which will be not be kept permanently open, so that it could be read with a text editor or a program while the server is still running.  ( Redirecting the console output to a file with `slingbox_server.exe > mylog.txt` is not OK as it does't allow seeing what the console displays.)

**Comment by GerryDazoo** [comment]
You've got two copies of the server code running at the same time.

Use the "tee" command to pipe std out to the console and a file.

**Reply by bradut** [reply]
Thank you Gerry,
I found a Windows port of `tee` [here](https://code.google.com/archive/p/wintee/).
Tried it in Windows: `slingbox_server.exe | wtee log1.txt` and seems to work fine.
